rest/server_context.go

```diff
@@ -82,6 +83,27 @@ func NewServerContext(config *ServerConfig) *ServerContext {
 		sc.startStatsReporter()
 	}
 
+	if config.Replications != nil {
+		go func() {
+			time.Sleep(kServerStartupWait)
```

Comment:
RunServer never returns it blocks on serving the Public REST API.

We could start replications after the ADMIN API is served, but this is run in a goroutine, so we don't know when endpoints are actually available. There is still likely to be a short delay.

And this won't help for the replications that use authenticated users via the public REST API for replications.

I can't see any notification mechanism to tell a client that an HTTP listener service is up.

We could change one shot replications to retry connections in the event of a failure.